### PR TITLE
Add WithoutStrictOrderingFor overload with an expression

### PR DESCRIPTION
**Pull request by chrischu**
I hope it's okay that I circumvented the "separate issue with api-approved label" since the feature I'm adding is essentially already there for "WithStrictOrderingFor".

### CODE REVIEW

File: Src/FluentAssertions/Equivalency/Ordering/PathBasedOrderingRule.cs [hunk 1]
```diff
@@ -16,6 +16,8 @@ public PathBasedOrderingRule(string path)
         this.path = path;
     }
 
+    public bool Invert { get; set; }
```
Comment (lg2de) [inline Src/FluentAssertions/Equivalency/Ordering/PathBasedOrderingRule.cs:19]: `Strict` ?

Comment (jnyrup) [reply Src/FluentAssertions/Equivalency/Ordering/PathBasedOrderingRule.cs:19]: `Invert` matches how it's named in `PredicateBasedOrderingRule`.
https://github.com/fluentassertions/fluentassertions/blob/81d3fbf7154f8d9cf80ca0fa72759f863132cf94/Src/FluentAssertions/Equivalency/Ordering/PredicateBasedOrderingRule.cs#L17

File: Src/FluentAssertions/Equivalency/EquivalencyAssertionOptions.cs [hunk 2]
```diff
@@ -74,6 +74,21 @@ public EquivalencyAssertionOptions<TExpectation> Including(Expression<Func<TExpe
         return this;
     }
 
+    /// <summary>
+    /// Causes the collection identified by <paramref name="expression"/> to be compared in the order
+    /// in which the items appear in the expectation.
```
Comment (jnyrup) [inline Src/FluentAssertions/Equivalency/EquivalencyAssertionOptions.cs:79]:
```suggestion
    /// Causes the collection identified by <paramref name="expression"/> to be compared ignoring the order
    /// in which the items appear in the expectation.
```

**Review by jnyrup — APPROVED** [review]
Thanks for adding this missing overload.
Please add a line to the [release notes](https://github.com/fluentassertions/fluentassertions/blob/develop/docs/_pages/releases.md)

Apparently I've known this shortcoming this since https://github.com/fluentassertions/fluentassertions/pull/974#issuecomment-441101713 but forgot to to anything about it.

**Comment by jnyrup** [comment]
Code Coverage points out that we need a failing test for this to hit both cases of `Invert` in `ToString()`

**Comment by chrischu** [comment]
> Code Coverage points out that we need a failing test for this to hit both cases of `Invert` in `ToString()`

I tried changing that but could not figure out where the coverage for the `Invert` property of `PredicateBasedOrderingRule` comes from (because I thought I could use that as a guide). 

It might even be impossible to cover the change, as the rule will only ever throw if `Invert` is `false`. Maybe the fix would be actually not vary the message based on the `Invert` property? I'll implement that change :).

File: Src/FluentAssertions/Equivalency/Ordering/PathBasedOrderingRule.cs [hunk 3]
```diff
@@ -60,6 +63,7 @@ private string RemoveInitialIndexQualifier(string sourcePath)
 
     public override string ToString()
```
Comment (jnyrup) [inline Src/FluentAssertions/Equivalency/Ordering/PathBasedOrderingRule.cs:64]: Something like this should hits the path.
```cs
[Fact]
public void Negated()
{
    // Arrange
    var subject = new
    {
        UnorderedCollection = new[] { 1 }
    };

    var expectation = new
    {
        UnorderedCollection = new[] { 2 }
    };

    // Act
    Action action = () => subject.Should().BeEquivalentTo(expectation, options => options
        .WithStrictOrdering()
        .WithoutStrictOrderingFor(x => x.UnorderedCollection));

    // Assert
    action.Should().Throw<XunitException>()
        .WithMessage("*not strict*");
}
```

File: Src/FluentAssertions/Equivalency/Ordering/PathBasedOrderingRule.cs (hunk 1 above)
Comment (dennisdoomen) [inline Src/FluentAssertions/Equivalency/Ordering/PathBasedOrderingRule.cs:19]: ♻️ You can change the setter to `init`

**Review by dennisdoomen — APPROVED** [review]
Only a suggestion from my IDE.

File: Src/FluentAssertions/Equivalency/Ordering/PathBasedOrderingRule.cs (hunk 1 above)
Comment (jnyrup) [reply Src/FluentAssertions/Equivalency/Ordering/PathBasedOrderingRule.cs:19]: I'll fix these in another PR

**Pull request merged**